src/players/PhenixPlayer.js

```diff
@@ -0,0 +1,215 @@
+// TODO: ReactPlayer's listener logic is very shaky because if you change the function identity
+//       it won't get cleaned up. This is an existing problem so we're not gonna fix it here.
+import React, { Component } from 'react'
+
+import { getSDK } from '../utils'
+import createSinglePlayer from '../singlePlayer'
+
+const PHENIX_SDK_URL = 'https://unpkg.com/phenix-web-sdk@2019.2.3/dist/phenix-web-sdk.min.js'
+const PHENIX_SDK_GLOBAL = 'phenix-web-sdk'
+
+// TODO: Add optional auth data parameter at the end
+const PHENIX_URL_REGEX = /^phenix:(.+?)\|(.+?)(?:\|(.+?))?$/i // i hate this so much
+
+function getPhenixSdk () {
+  return getSDK(PHENIX_SDK_URL, PHENIX_SDK_GLOBAL)
+}
+
+function canPlay (url) {
+  return PHENIX_URL_REGEX.test(url)
+}
+
+export class PhenixPlayer extends Component {
+  static displayName = 'PhenixPlayer'
+  static canPlay = canPlay
+  player = null
+  channelExpress = null
+
+  playerRef = (player) => {
+    if (player === this.player) {
+      return;
+    }
+    if (this.player) {
+      this.removeListeners()
+    }
+    this.player = player
+    if (this.player) {
+      this.addListeners()
+    }
+  }
+
+  componentWillUnmount () {
+    // TODO: If refs get called with null on unmount, no reason to do this
+    if (this.player) {
+      this.removeListeners()
+      this.player = null
+    }
+    if (this.channelExpress) {
+      this.channelExpress.dispose();
+      this.channelExpress = null;
+    }
+  }
+
+  addListeners () {
+    const { onReady, onPlay, onPause, onEnded, onVolumeChange, onError, playsinline, videoElementId } = this.props
+    this.player.addEventListener('canplay', onReady)
+    this.player.addEventListener('play', onPlay)
+    this.player.addEventListener('pause', onPause)
+    this.player.addEventListener('seeked', this.onSeek)
+    this.player.addEventListener('ended', onEnded)
+    this.player.addEventListener('error', onError)
+    this.player.addEventListener('volumechange', onVolumeChange)
+    // wow
```

Comment:
lmao

Reply:
that's actually copypasta'd from the file player LMAO